rust/crd/src/lib.rs

```diff
@@ -384,19 +385,23 @@ impl HdfsCluster {
 /// Reference to a single `Pod` that is a component of a [`HdfsCluster`]
 ///
 /// Used for service discovery.
+#[derive(Clone, Debug)]
 pub struct HdfsPodRef {
     pub namespace: String,
     pub role_group_service_name: String,
     pub pod_name: String,
     pub ports: HashMap<String, i32>,
+    pub external_name: Option<String>,
 }
 
 impl HdfsPodRef {
     pub fn fqdn(&self) -> String {
-        format!(
-            "{}.{}.{}.svc.cluster.local",
-            self.pod_name, self.role_group_service_name, self.namespace
-        )
+        self.external_name.clone().unwrap_or_else(|| {
```

Comment:
@razvan would it make sense to split `HdfsPodRef` into separate types for journalnodes and namenodes? We'll probably want to only use the external name for *all* namenode traffic anyway, once we start doing Kerberos (since this is linked to the hostname, and HDFS nodes dislike having more than one Kerberos principal each).

Reply:
Makes sense.